source/numeric-range.d.ts

```diff
@@ -0,0 +1,68 @@
+type Infinities = 'Infinity' | '-Infinity';
+
+/**
+A `number` that is an integer.
+
+Use-case: Validating and documenting parameters.
+
+@example
+```
+declare function setYear<T extends number>(length: Integer<T>): void;
+```
+
+@category Utilities
+*/
+// `${bigint}` is a type that matches a valid bigint literal without the `n` (ex. 1, 0b1, 0o1, 0x1)
+// Because T is a number and not a string we can effectively use this to filter out any numbers containing decimal points
+export type Integer<T extends number> = `${T}` extends Infinities ? never : `${T}` extends `${bigint}` ? T : never;
```

Comment:
I wonder how this technique compares to https://stackoverflow.com/a/69413070/64949